spring-cloud-context/src/main/java/org/springframework/cloud/autoconfigure/RefreshEndpointAutoConfiguration.java

```diff
@@ -56,45 +56,45 @@ RefreshScopeHealthIndicator refreshScopeHealthIndicator(RefreshScope scope,
 		return new RefreshScopeHealthIndicator(scope, rebinder);
 	}
 
-	@ConditionalOnClass(IntegrationMBeanExporter.class)
-	protected static class RestartEndpointWithIntegration {
+	@ConditionalOnProperty(value = "endpoints.restart.enabled", matchIfMissing = true)
```

Comment:
OK, will change it, including the other places where @ConditionalOnProperty was already used.